pkg/controller/nginxingress/nginx_ingress_controller.go

```diff
@@ -560,3 +567,31 @@ func ToNginxIngressConfig(nic *approutingv1alpha1.NginxIngressController, defaul
 
 	return nginxIng
 }
+
+func getTargetCPUUtilizationPercentage(nic *approutingv1alpha1.NginxIngressController) int32 {
+	if nic == nil {
+		return defaultTargetCPUUtilization
+	}
+
+	scaling := nic.Spec.Scaling
```

Comment:
is it possible for nic.Spec to be nil/do we need a nil check for that as well

Reply:
no